src/controls/controlRegistry.ts

```diff
@@ -62,16 +85,44 @@ export class ControlRegistry {
   }
 }
 
+/**
+ * Creates an instance of IControlDescriptor based on information provided by the @implementProperty decorator
+ * on a control
+ */
 export class ReflectionBasedDescriptor<T extends Control> implements IControlDescriptor<T> {
-  constructor(public readonly id: string, private readonly typeDef: new () => T) {}
-  getProperties(): IControlProperty[] {
+  constructor(
+    public readonly id: string,
+    public readonly displayName: LocalizedString,
+    private readonly typeDef: new () => T,
+    private readonly defaultValuesCreator?: () => IDefaultControlValues,
+  ) {}
+
+  public getProperties(): IControlProperty[] {
     return getControlPropertiesFor(this.typeDef.prototype) ?? [];
   }
 
   public createInstance(): T {
     return new this.typeDef();
   }
 
+  public getDefaultValues(): Required<IDefaultControlValues> {
+    let defaultValues = this.defaultValuesCreator?.() ?? {};
+
+    let properties = defaultValues?.properties ?? {};
+    let position = defaultValues?.position ?? {};
+
+    let textId = 'text.text';
```

Comment:
I'm not a fan of hardcoding this - can we instead pull-in the TextContent property?
Does that result in a circular loop?

Reply:
I started down the path of doing this, but it turned into a general refactoring of how we reference this.  I'll do this in a follow up